### PR TITLE
Implement running individual packages from a project

### LINKED ISSUE #371: Support specifying a single package to build on a project with `garn build project.package`

### PR DESCRIPTION
**Pull request by soenkehahn**
Fixes #371.

### CODE REVIEW

File: src/Garn/GarnConfig.hs [hunk 1]
```diff
@@ -78,6 +78,7 @@ type Targets = Map TargetName TargetConfig
 
 data TargetConfig
   = TargetConfigProject ProjectTarget
+  | TargetConfigPackage PackageTarget
```
Comment (soenkehahn) [inline src/Garn/GarnConfig.hs:81]: This is an IPC-breaking change.

File: test/spec/BuildSpec.hs [hunk 2]
```diff
@@ -3,6 +3,7 @@
 module BuildSpec where
 
 import Data.String.Interpolate (i)
+import Data.String.Interpolate.Util (unindent)
```
Comment (soenkehahn) [inline test/spec/BuildSpec.hs:6]: Please review this file with hidden whitespace changes!

File: test/spec/BuildSpec.hs [hunk 3]
```diff
@@ -11,105 +12,166 @@ import Test.Mockery.Directory
...
+              );
+            |]
+        output <- runGarn ["build", "project.foo"]
+        exitCode output `shouldBe` ExitSuccess
+        readFile "result/build-artifact" `shouldReturn` "foo\n"
+        output <- runGarn ["build", "project.bar"]
+        exitCode output `shouldBe` ExitSuccess
+        readFile "result/build-artifact" `shouldReturn` "bar\n"
+
+      it "builds top-level packages" $ \runGarn -> do
```
Comment (soenkehahn) [inline test/spec/BuildSpec.hs:77]: This wasn't required by #391, but it was easier to support this than not.

File: ts/internal/runner.ts [hunk 4]
```diff
@@ -238,7 +246,7 @@ const collectProjectChecks = (
   project: Project,
 ): Record<string, Check> =>
   mapKeys(
-    (name) => `${projectName}_${name}`,
+    (name) => `${projectName}/${name}`,
```
Comment (soenkehahn) [inline ts/internal/runner.ts:249]: This change is unrelated but made for consistency.

File: ts/package.ts [hunk 5]
```diff
@@ -27,7 +25,7 @@ export function isPackage(x: unknown): x is Package {
  */
 export function mkPackage(
   nixExpression: NixExpression,
-  description?: string,
+  description: string,
```
Comment (soenkehahn) [inline ts/package.ts:28]: This is a breaking change of the typescript library. I do think that it's nicer to make the description be required.

File: CHANGELOG.md [hunk 6]
```diff
@@ -1,5 +1,10 @@
 # Changelog
 
+## v0.0.16
```
Comment (alexdavid) [inline CHANGELOG.md:3]:
```suggestion
## v0.0.16 (unreleased)
```

Maybe part of release process is to remove `(unreleased)` and add the next version with it to make it more obvious when it is ok to add new features to a version without checking which version is currently released?

File: test/spec/BuildSpec.hs [hunk 7]
```diff
@@ -11,105 +12,166 @@ import Test.Mockery.Directory
...
+                  longer: garn.build`
+                    # this is some longer build script:
+                    bla bla bla
+                  `,
+                },
+              );
+            |]
+        output <- runGarn ["build"]
+        stderr output `shouldContain` "project.short            Builds short command"
+        stderr output `shouldContain` "project.longer           Builds # this is some longe..."
```
Comment (alexdavid) [inline test/spec/BuildSpec.hs:108]: TIOLI, just felt like a change of number of spaces between the name and description shouldn't break this test
```suggestion
        stderr output `shouldMatch` "project.short \\s+ Builds short command"
        stderr output `shouldMatch` "project.longer \\s+ Builds # this is some longe\\.\\.\\."
```

(The extra spaces around `\\s+` make it a bit easier to read, and will only fail if there are fewer than 3 spaces which felt reasonable to me)

**Review by alexdavid — APPROVED** [review]
Looks great! I'm super excited for this!

File: CHANGELOG.md (hunk 6 above)
Comment (soenkehahn) [reply CHANGELOG.md:3]: Yeah, that sounds right. Also, we want the changelog file on github to be explicit about what's released and what's not. For users, I mean.

**Pull request merged**